lib/transaction.py

```diff
@@ -909,14 +935,14 @@ def _ecdsa_sign(sec, pre_hash):
         return sig
 
     @staticmethod
-    def _schnorr_sign(pubkey, sec, pre_hash):
+    def _schnorr_sign(pubkey, sec, pre_hash, ndata=None): 
```

Comment:
I wonder if we should make this `ndata` argument a "kw-only arg" -- those are arguments after the `*, ` and they can only be specified using keywords.  It's a good way to add arguments to existing functions for stuff that is not considered a "primary feature" of the function.  In this case I think the `ndata` qualifies as such an argument.  So you'd change the call signature here to:

```python
def _schnorr_sign(pubkey, sec, pre_hash, *, ndata=None):
```

And then when you call it later in this file, you'd call it as:
```python
    sig = self._schnorr_sign(pubkey, sec, pre_hash, ndata=ndata)
```